ReSharper.FSharp/src/FSharp.Psi.Features/src/Util/FSharpMethodInvocationUtil.fs

```diff
@@ -1,44 +1,85 @@
 module JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Util.FSharpMethodInvocationUtil
 
+open FSharp.Compiler.SourceCodeServices
 open JetBrains.ReSharper.Plugins.FSharp.Psi
 open JetBrains.ReSharper.Plugins.FSharp.Psi.Impl
 open JetBrains.ReSharper.Plugins.FSharp.Psi.Tree
 open JetBrains.ReSharper.Psi
+open JetBrains.ReSharper.Psi.ExtensionsAPI
 open JetBrains.ReSharper.Psi.Tree
 
-let tryGetNamedArg (expr: IFSharpExpression) =
-    let binaryAppExpr = expr.As<IBinaryAppExpr>()
-    if isNull binaryAppExpr then null else
 
+let resolveNamedArg (binaryAppExpr: IBinaryAppExpr) =
     if binaryAppExpr.Operator.Reference.GetName() <> "=" then null else
 
     let refExpr = binaryAppExpr.LeftArgument.As<IReferenceExpr>()
     if isNull refExpr then null else
 
     refExpr.Reference.Resolve().DeclaredElement.As<IParameter>()
 
-let getMatchingParameter (initialExpr: IFSharpExpression) =
-    let expr = initialExpr.IgnoreInnerParens()
-    let tupleExpr = TupleExprNavigator.GetByExpression(expr)
-    let tupleExprContext = tupleExpr.IgnoreParentParens()
 
-    let appExpr = PrefixAppExprNavigator.GetByArgumentExpression(if isNull tupleExpr then expr else tupleExprContext)
-    if isNull appExpr then null else
+let tryGetNamedArg (expr: IFSharpExpression) =
+    let binaryAppExpr = expr.As<IBinaryAppExpr>()
+    if isNull binaryAppExpr then null else
+    resolveNamedArg binaryAppExpr
```

Comment:
Since binaryAppExpr is the only path we're interested in (and we don't need to fall shortly or continue) it's a bit cleaner to use a normal `match` here.
```suggestion
    match expr with
    | :? IBinaryAppExpr as binaryAppExpr -> resolveNamedArg binaryAppExpr
    | _ -> null
```